src/lib.rs

```diff
@@ -230,4 +256,12 @@ macro_rules! impl_Integer128 {
 }
 
 #[cfg(all(feature = "i128"))]
-impl_Integer128!(i128, u128);
+const U128_MAX_LEN: usize = 39;
+#[cfg(all(feature = "i128"))]
+const I128_MAX_LEN: usize = 40;
+
+#[cfg(all(feature = "i128"))]
+impl_Integer!(private U128_MAX_LEN => u64, I128_MAX_LEN => u64 as u64);
```

Comment:
I think rather than having 3 different implementations of IntegerPrivate for u64 for the three different array lengths, we should have just IntegerPrivate<[u8; U64_MAX_LEN]> and have casts or whatever in the 128-bit impls to leverage that one u64 impl. Is it possible to make that work?